File: week_1/task_a.cc
#include <climits>
#include <iostream>
#include <vector>

int main() {
  int vasya_sum = 0, masha_sum = 0;
  int masha_max = 0, vasya_min = INT_MAX;

  int n = 0;
  std::cin >> n;
  std::vector<int> a(n);

  for (int i = 0; i < n; ++i) {
    std::cin >> a[i];
  }

  for (int i = 0; i < n; ++i) {
    if (i % 2 == 0) {
      vasya_sum += a[i];
      vasya_min = std::min(vasya_min, a[i]);
    } else {
      masha_sum += a[i];
      masha_max = std::max(masha_max, a[i]);
    }
  }

  if (vasya_min < masha_max) {
    vasya_sum += (masha_max - vasya_min);
    masha_sum += (vasya_min - masha_max);
  }

  std::cout << vasya_sum - masha_sum;

  return 0;
}

File: week_1/go/task_a.go
package main

import (
	"bufio"
	"os"
	"strconv"
	"strings"
)

func main() {
	var (
		vasya_sum int
		masha_sum int
		masha_max int
		vasya_min int
	)

	reader := bufio.NewReaderSize(os.Stdin, 1<<20)
	writer := bufio.NewWriterSize(os.Stdout, 1<<20)
	defer writer.Flush()

	line, _ := reader.ReadString('\n')
	n, _ := strconv.Atoi(strings.TrimSpace(line))

	a := make([]int, n)

	line, _ = reader.ReadString('\n')
	numbers := strings.Fields(line)

	vasya_min = 1 << 30

	for i := 0; i < n; i++ {
		a[i], _ = strconv.Atoi(numbers[i])
	}

	for i := 0; i < n; i++ {
		if i%2 == 0 {
			vasya_sum += a[i]
			if a[i] < vasya_min {
				vasya_min = a[i]
			}
		} else {
			masha_sum += a[i]
			if a[i] > masha_max {
				masha_max = a[i]
			}
		}
	}

	if vasya_min < masha_max {
		vasya_sum += (masha_max - vasya_min)
		masha_sum += (vasya_min - masha_max)
	}

	result := vasya_sum - masha_sum

	writer.WriteString(strconv.Itoa(result))
	writer.WriteByte('\n')

}
